bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/ScalarWebAvContentProtocol.java

```diff
@@ -0,0 +1,3223 @@
+                    "scalarwebavcontrolcontentprogramtitle"));
+            // descriptors.add(createDescriptor(createChannel(CN_RECORDINGINFO), "String",
+            // "scalarwebavcontrolcontextrecordingInfo")); // todo
+            descriptors.add(createDescriptor(createChannel(CN_REMOTEPLAYTYPE), "String",
+                    "scalarwebavcontrolcontentremoteplaytype"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_REPEATTYPE), "String", "scalarwebavcontrolcontentrepeattype"));
+            descriptors.add(createDescriptor(createChannel(CN_SERVICE), "String", "scalarwebavcontrolcontentservice"));
+            descriptors.add(createDescriptor(createChannel(CN_SIZEMB), "Number", "scalarwebavcontrolcontentsizemb"));
+            descriptors.add(createDescriptor(createChannel(CN_SOURCE), "String", "scalarwebavcontrolcontentsource"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_SOURCELABEL), "String", "scalarwebavcontrolcontentsourcelabel"));
+            descriptors.add(createDescriptor(createChannel(CN_STARTDATETIME), "String",
+                    "scalarwebavcontrolcontentstartdatetime"));
+            descriptors.add(createDescriptor(createChannel(CN_STATE), "String", "scalarwebavcontrolcontentstate"));
+            descriptors.add(createDescriptor(createChannel(CN_STATESUPPLEMENT), "String",
+                    "scalarwebavcontrolcontentstatesupplement"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_STORAGEURI), "String", "scalarwebavcontrolcontentstorageuri"));
+            descriptors.add(createDescriptor(createChannel(CN_SUBTITLELANGUAGE), "String",
+                    "scalarwebavcontrolcontentsubtitlelanguage"));
+            descriptors.add(createDescriptor(createChannel(CN_SUBTITLETITLE), "String",
+                    "scalarwebavcontrolcontentsubtitletitle"));
+            descriptors.add(createDescriptor(createChannel(CN_SYNCCONTENTPRIORITY), "String",
+                    "scalarwebavcontrolcontentsynccontentpriority"));
+            descriptors.add(createDescriptor(createChannel(CN_TITLE), "String", "scalarwebavcontrolcontenttitle"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_TOTALCOUNT), "Number", "scalarwebavcontrolcontenttotalcount"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_TRIPLETSTR), "String", "scalarwebavcontrolcontenttripletstr"));
+            descriptors.add(createDescriptor(createChannel(CN_USERCONTENTFLAG), "Switch",
+                    "scalarwebavcontrolcontentusercontentflag"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_VIDEOCODEC), "String", "scalarwebavcontrolcontentvideocodec"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_VISIBILITY), "String", "scalarwebavcontrolcontentvisibility"));
+        }
+
+    }
```

Comment:
This method has a ton of code duplication

Reply:
Yes - a bit of a coding call on my part.  The issue is that each version can change something on the prior version (sometimes a boolean becomes a string [or in reverse], sometimes they drop a field, the always add new fields, etc).  So I can either figure out every change in every version and code that up to prevent duplicates or simply show which each version has (and have duplicates).  Ultimately I decided it would be less error prone, more simple and easier to read to show each version.  A bit of a call on my part..